express-notes-api/index.js

```diff
@@ -0,0 +1,165 @@
+const express = require('express');
+const fs = require('fs');
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const data = require('./data.json');
+  const arr = [];
+
+  for (const entry in data.notes) {
+    arr.push(data.notes[entry]);
+  };
+  res.status(200).send(arr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const data = require('./data.json');
+  const id = req.params.id;
+  let found = false;
+
+  if (id >= 1) {
+    for (const entry in data.notes) {
```

Comment:
Instead of using a for loop to check every property one at a time, simply check if there is a property in the data which corresponds to the supplied id:
```JAVASCRIPT
if(notes[id]) {
  // Note exists, send it to the client
} else {
  // Note does not exist, send a 404 error to the client
}
```